### ISSUE TITLE
How to provide default value to StrictBool?

### ISSUE THREAD

**Issue by pykong** [samuelcolvin/pydantic#690]
How can we provide a default value to a variable with type `StrictBool` in a way that is accepted by MyPy?

```py
from pydantic import BaseModel

class Flags(BaseModel):
    pretend: StrictBool = False  # MyPy does not like this
```

**Comment by dmontagu** [comment]
Booleans are hard to deal with in a type-hinting context because python explicitly disallows subclassing bool. I might be missing something, but I suspect there isn't a way that you can assign a default value without having to fight mypy at least a little.

One option would be to just use
```python
class Flags(BaseModel):
    pretend: StrictBool = False  # type: ignore
```
but that may cause type checking issues since `StrictBool` is (by necessity) a subclass of int, rather than bool.

Another option is to put it in an `if TYPE_CHECKING:` block:
```python
from typing import TYPE_CHECKING

from pydantic import BaseModel

if TYPE_CHECKING:
    StrictBool = bool
else:
    from pydantic import StrictBool

class Flags(BaseModel):
    pretend: StrictBool = False
```

This should behave properly with type checking, but is a little uglier everywhere you import it.

@samuelcolvin One approach might be to put the above alias for StrictBool (and RelaxedBool if/when merged) directly into pydantic. (Or perhaps using a `types.pyi` or similar?) Any obvious downsides to that?

**Comment by samuelcolvin** [comment]
> @samuelcolvin Do you think it could make sense to just put the above alias for StrictBool (and RelaxedBool if/when merged) directly into pydantic? Or perhaps into a types.pyi or similar? Any obvious downsides to that?

Makes sense. 

Although my instinct is that the need for `StrictBool` will reduce a lot when #617 gets deployed.

**Comment by pykong** [comment]
Thanks for your responses! IMHO a stricter `bool` validation as implemented in #617 would be the cleanest solution. Until it is merged and released I can fall back on suppressing `MyPy` warnings of `bool` and `Strictbool` type mismatch.
